src/operator/effectivepolicy/Syncer.go

```diff
@@ -0,0 +1,142 @@
+package effectivepolicy
+
+import (
+	"context"
+	goerrors "errors"
+	"github.com/amit7itz/goset"
+	"github.com/otterize/intents-operator/src/operator/api/v1alpha3"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/otterize/intents-operator/src/shared/serviceidresolver/serviceidentity"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type EffectivePoliciesApplier interface {
+	ApplyEffectivePolicies(ctx context.Context, eps []ServiceEffectivePolicy) (int, error)
```

Comment:
Since the appliers behave like reconcilers (they have all the same properties - supposed to bring the cluster to a desired state, idempotent, etc), let's use the same word (Reconcile?)

If it means the same thing it's better to use the same terminology